From cf540956f8bc4e94fd7c7f361441a6c0325cb5cc Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 29 Jan 2026 13:32:42 +1100
Subject: [PATCH 06/11] skip metal disassembly for vanilla clang

---
 tinygrad/runtime/ops_metal.py | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/tinygrad/runtime/ops_metal.py b/tinygrad/runtime/ops_metal.py
index da8295738..28f97d28b 100644
--- a/tinygrad/runtime/ops_metal.py
+++ b/tinygrad/runtime/ops_metal.py
@@ -104,7 +104,18 @@ class MetalCompiler(Compiler):
     if isinstance(ret, Exception): raise ret
     assert ret[:4] == b"MTLB" and ret[-4:] == b"ENDT", f"Invalid Metal library. {ret!r}"
     return ret
+  @functools.cache
+  def is_apple_clang(self) -> bool:
+    try:
+      p = subprocess.run(["clang", "-dM", "-E", "-"], input="", text=True, capture_output=True, timeout=2)
+    except Exception:
+      return False
+    return "__apple_build_version__" in p.stdout
   def disassemble(self, lib:bytes):
+    if not self.is_apple_clang():
+      # only appleclang generates the device-specific binary that can be
+      # disassembled, see https://github.com/tinygrad/tinygrad/issues/14005
+      return
     with tempfile.NamedTemporaryFile(delete=True) as shader:
       shader.write(lib)
       shader.flush()
